lib/MusicBrainz/Server/Data/Relationship.pm

```diff
@@ -551,6 +556,7 @@ sub get_types_for_timeline
         }
     }
 
+    $cache->set('types_for_timeline', \%rel_types);
```

Comment:
I guess we should set a timeout here in case it's not evicted before 24 hours or so.  Does that seem like a reasonable amount of time?
```suggestion
    $cache->set('types_for_timeline', \%rel_types, 60 * 60 * 24);
```

Reply:
Either that, or it should just never expire and then be expired by hand (well, by Perl) when editing or adding a relationship type, since that often happens rarely (there might be entire months where nothing changes)

Reply:
I was going to say that expiring by hand might be annoying since the `get_types_for_timeline` query also touches the `link` table to get a count of rels.  But looking at the query again I'm not sure why it needs a count, or why it joins the link table at all if it only uses `$_->{name}` below.  Can you explain how that works?

Reply:
... I probably took that from the `count.ar.links.table.type_name` query I started from, and never took it out, which does seem dumb, now that you point it out. Wonder how much faster the whole thing will be if I drop that.

Reply:
A lot. The query needed to be like 1/5th of what it was right now. Updated the PR with that in another commit.